zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/UserTaskCreatingApplier.java

```diff
@@ -28,7 +29,14 @@ public UserTaskCreatingApplier(final MutableProcessingState processingState) {
 
   @Override
   public void applyState(final long key, final UserTaskRecord value) {
-    userTaskState.create(value);
+    // todo: should we actually store the user task fully already, or wait until we finalize?
+    final var copy = value.copy();
+    copy.unsetAssignee();
+    userTaskState.create(copy);
```

Comment:
💭  This is necessary to ensure the UserTaskRecord still has the assignee set, which is used after creation in the task listener (to set `assignee` property in custom headers)